node/src/accountant/payment_adjuster/preparatory_analyser/mod.rs

```diff
@@ -495,171 +495,212 @@ mod tests {
-            PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+        let ensure_accounts_right_type = |accounts| accounts;
+        let prepare_expected_error = |number_of_accounts, _, cw_service_fee_balance_minor| {
+            PaymentAdjusterError::AbsolutelyInsufficientServiceFeeBalancePostTxFeeAdjustment {
                 original_number_of_accounts,
                 number_of_accounts,
-                original_service_fee_required_total_minor: initial_sum,
+                original_total_service_fee_required_minor: initial_sum,
                 cw_service_fee_balance_minor,
             }
         };
 
-        test_not_enough_for_even_the_least_demanding_account_causes_error(
+        test_not_enough_even_for_the_smallest_account_error(
             error_factory,
-            expected_error_preparer,
+            ensure_accounts_right_type,
+            prepare_expected_error,
         )
     }
 
     #[test]
-    fn accounts_analyzing_works_even_for_weighted_payable() {
+    fn recheck_if_service_fee_adjustment_is_needed_works_nicely_for_weighted_payables() {
         init_test_logging();
-        let test_name = "accounts_analyzing_works_even_for_weighted_payable";
-        let balance_1 = multiple_by_billion(2_000_000);
-        let mut weighted_account_1 = make_weighed_account(123);
-        weighted_account_1
+        let test_name =
+            "recheck_if_service_fee_adjustment_is_needed_works_nicely_for_weighted_payables";
+        let balance_1 = multiply_by_billion(2_000_000);
+        let mut weighed_account_1 = make_meaningless_weighed_account(123);
+        weighed_account_1
             .analyzed_account
             .qualified_as
             .bare_account
             .balance_wei = balance_1;
-        let balance_2 = multiple_by_billion(3_456_000);
-        let mut weighted_account_2 = make_weighed_account(456);
-        weighted_account_2
+        let balance_2 = multiply_by_billion(3_456_000);
+        let mut weighed_account_2 = make_meaningless_weighed_account(456);
+        weighed_account_2
             .analyzed_account
             .qualified_as
             .bare_account
             .balance_wei = balance_2;
-        let accounts = vec![weighted_account_1, weighted_account_2];
+        let accounts = vec![weighed_account_1, weighed_account_2];
         let service_fee_totally_required_minor = balance_1 + balance_2;
+        // We start at a value being one bigger than required, and in the act, we subtract from it
+        // so that we also get the exact edge and finally also not enough by one.
         let cw_service_fee_balance_minor = service_fee_totally_required_minor + 1;
         let error_factory = LateServiceFeeSingleTxErrorFactory::new(&accounts);
         let logger = Logger::new(test_name);
         let subject = PreparatoryAnalyzer::new();
 
         [(0, false), (1, false), (2, true)].iter().for_each(
```

Comment:
There was an idea presented by you that would eliminate the need to remove the comment.

You'd like to use this instead:

`0` -> `service_fee_totally_required_minor - 1`

Similarly, you'd replace `1` and `2`.

Good Idea  👍